src/logger.c

```diff
@@ -90,17 +96,31 @@ static struct log_file pot = {NULL, NULL, NULL, 0, -1};
 
 static int in_logger = 0;
 
-static void log_file_init(struct log_file *f, char *name, int size)
+static void log_file_init(struct log_file *f, char *name, char *perms, int size)
 {
+	p_u = "0600";
+	p_ug = "0660";
+	p_ugo = "0666";
+	if (!((strcmp(perms, p_u) == 0) || (strcmp(perms, p_ug) == 0) || (strcmp(perms, p_ugo) == 0)))
+		pexit("PotFilePerms or LogFilePerms %s invalid, valid values: %s, %s, %s\n\n", perms, p_u, p_ug, p_ugo);
+	perms_t = strtoul(perms, NULL, 8);
```

Comment:
@jfoug Thanks! I will try testing what happens after conversion.  The issue I was having is that if a user specified for example 666 instead of 0666 it would set sticky bit and be like 400 for the rest of the perms or something weird like that.  I might have been passing the param as an int at the time and not a ulong, which was dropping my preceding zero no matter what. I'll have to test and see.  Besides that, much simpler thank you.